src/dscontrib/sustainability/utils.py

```diff
@@ -0,0 +1,58 @@
+import pandas as pd
+import pyspark.sql.functions as F
+
+
+def bootstrap(df, iterations, reporting_freq=10, strata=[],
+              treatment='variant', control='control'):
+    def munge(s):
+        agg = s.groupby(["experiment_branch"] + strata).mean()
+        agg = (
+          pd.melt(agg.reset_index(), id_vars=["experiment_branch"] + strata)
+          .pivot_table(
+                    index=strata + ['variable'],
+                    columns='experiment_branch',
+                    values='value').reset_index()
+          )
+        agg['change'] = (agg[treatment] - agg[control]) / agg[control]
+        return agg
+
+    n = len(df)
+    results = munge(df)
+    results['iteration'] = -1  # true sample stats lie at -1
+    for i in range(iterations):
+        if i % 10 == 0:
+            print i,
+        s = df.sample(n, replace=True)
+        sagg = munge(s)
+        sagg['iteration'] = i
+        results = pd.concat([sagg, results])
+    return results
+
+
+measures = {
```

Comment:
I think these limits should either be sanity checks or (dynamically generated) percentile filters. If sanity checks, I think these limits should be a little looser.